corehq/sql_db/util.py

```diff
@@ -86,38 +45,38 @@ def paginate_query_across_partitioned_databases(model_class, q_expression, annot
     and their calculations. The dictionary will be splatted into the `.annotate` function
 
     :param values: (optional) If specified, should be a list of values to retrieve rather
-    than retrieving entire objects. If `pk` is not defined in `values`, then the values
-    returned will be a tuple of (pk + *values)
+    than retrieving entire objects.
 
     :return: A generator with the results
     """
     db_names = get_db_aliases_for_partitioned_query()
     sort_col = 'pk'
 
     return_values = None
-    if values and sort_col not in values:
-        return_values = ['pk'] + values
+    if values:
+        return_values = [sort_col] + values
 
     for db_name in db_names:
         qs = model_class.objects.using(db_name)
         if annotate:
             qs = qs.annotate(**annotate)
 
         qs = qs.filter(q_expression)
-        value = 0
         last_value = qs.order_by('-{}'.format(sort_col)).values_list(sort_col, flat=True).first()
         if last_value is not None:
             qs = qs.order_by(sort_col)
+            value = qs.order_by(sort_col).values_list(sort_col, flat=True).first()
```

Comment:
The `order_by(sort_col)` on this line is redundant since it's added on the line above, right?

Reply:
```suggestion
            value = qs.values_list(sort_col, flat=True).first()
```